src/bullets/data_source/data_source_fmp.py

```diff
@@ -12,64 +16,88 @@ def __init__(self, token: str, resolution: Resolution):
         super().__init__()
         self.token = token
         self.resolution = resolution
-        self.stocks = {}
 
-    def get_historical_daily_prices(self, symbol: str, start_date: date, end_date: date = None):
+    def get_prices(self, symbol: str, start_timestamp: datetime, end_timestamp: datetime = None, delta: int = None,
+                   value: str = None):
         """
-        Gets the historical prices of the given stock for the given interval period
+        Gets the prices of the given stock for the given interval period
+        Can be between two dates or from a delta timeframe
         Args:
             symbol: Symbol of the stock/forex
-            start_date: Starting time of the interval of historical prices
-            end_date: Ending time of the interval of historical price. Default value is the time of the backtest
+            start_timestamp: Starting time of the interval wanted
+            end_timestamp: Ending time of the interval wanted
+            delta: Time delta from the start_timestamp wanted
+            value: Value of the stock you want, default is close
         Returns: An array of the closing price of the stock for every day between the interval
         """
-        if end_date is None:
-            end_date = self.timestamp.date()
-
-        if symbol not in self.stocks:
-            self._store_price_points(symbol, start_date, end_date)
+        if start_timestamp is None:
+            start_date = self.timestamp
         else:
-            stock = self.stocks[symbol]
+            start_date = start_timestamp
+
+        if end_timestamp is None and delta is None:
+            raise ValueError("End timestamp and delta cannot be None, one or the other has to be entered.")
```

Comment:
Should we maybe have a default delta of 1?

Reply:
True, changed